### PR TITLE
[Refax] : Clean up: declare all global interfaces #65

### LINKED ISSUE #65: [Feat] : Clean up: declare all global interfaces
**Is your feature request related to a problem? Please describe.**
Clean up and Declare all important interfaces

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by ballyalley-o**
This pull request includes several changes to clean up and declare all global interfaces. The changes include refactoring code to remove unused imports, updating middleware to include additional fields in models, removing unused interface files, and adding interface files for Mongoose schema and models. These changes improve code organization and maintainability. Fixes #65

### CODE REVIEW

**Comment by ballyalley-o** [comment]
For your review: @in-a-iii

File: src/interface/middleware.ts [hunk 1]
```diff
@@ -10,16 +8,6 @@ export interface IExpressController {
   (req: IRequestExtended, res: Response, next: NextFunction): void
 }
```
Comment (in-a-iii) [inline src/interface/middleware.ts:9]: use d.ts as an ext for consistency.

**Review by in-a-iii — CHANGES_REQUESTED** [review]

File: src/interface/utility.d.ts [hunk 2]
```diff
@@ -1,8 +1,8 @@
-export interface IConNex {
```
Comment (in-a-iii) [inline src/interface/utility.d.ts:1]: great, bout to say this one too.

File: src/util/connex.ts [hunk 3]
```diff
@@ -1,5 +1,3 @@
-import { IConNex } from '@interface/utility'
-
 const conNex: IConNex = (...param) => {
   return param.join('')
 }
```
Comment (in-a-iii) [inline src/util/connex.ts:3]: looks good. good to go. 👍

**Review by in-a-iii — APPROVED** [review]

File: src/util/connex.ts (hunk 3 above)
Comment (ballyalley-o) [reply src/util/connex.ts:3]: zooom

**Pull request merged**